test/infrastructure/inmemory/internal/server/api/handler.go

```diff
@@ -73,29 +73,33 @@ func NewAPIServerHandler(manager cmanager.Manager, log logr.Logger, resolver Res
 
 	// CRUD endpoints (global objects)
 	ws.Route(ws.POST("/api/v1/{resource}").Consumes(runtime.ContentTypeProtobuf).To(apiServer.apiV1Create))
-	ws.Route(ws.GET("/api/v1/{resource}").To(apiServer.apiV1List))
+	ws.Route(ws.GET("/api/v1/{resource}").If(isList).To(apiServer.apiV1List))
```

Comment:
Route handling for Watch and List this way seems more natural and easy to follow this way IMO.